common/src/allocations/monitor.rs

```diff
@@ -135,8 +46,61 @@ pub fn indexer_allocations(
     )
 }
 
+pub async fn get_allocations(
+    network_subgraph: &'static SubgraphClient,
+    indexer_address: Address,
+    recently_closed_allocation_buffer: Duration,
+) -> Result<HashMap<Address, Allocation>, anyhow::Error> {
+    let start = SystemTime::now();
+    let since_the_epoch = start
+        .duration_since(UNIX_EPOCH)
+        .expect("Time went backwards");
+    let closed_at_threshold = since_the_epoch - recently_closed_allocation_buffer;
+
+    let query = format!(
+        r#"
+            allocations(
+                block: $block
+                orderBy: id, orderDirection: asc
+                first: $first
+                where: {{
+                  and: [
+                    {{id_gt: $last}},
+                    {{indexer_: {{id: "{}" }} }}, {{or: [{{status: Active}}, {{and: [{{status: Closed, closedAt_gte:  {} }}] }}] }}
+                ]  
+                }}
+            ) {{
+                id
+                indexer {{
+                    id
+                }}
+                allocatedTokens
+                createdAtBlockHash
+                createdAtEpoch
+                closedAtEpoch
+                subgraphDeployment {{
+                    id
+                    deniedAt
+                }}
+            }}
+        "#,
+        indexer_address.to_string().to_ascii_lowercase(),
+        closed_at_threshold.as_secs(),
+    );
+
+    let responses = network_subgraph
+        .paginated_query::<Allocation>(query, 1000)
```

Comment:
Maybe using 200 would be more reasonable. It's the value that's used in the gateway code btw.